server/resp/src/test/java/org/infinispan/server/resp/RespSingleNodeTest.java

```diff
@@ -1008,15 +1021,31 @@ public void testSort() {
       // SORT people BY w_* GET o_*
       assertThat(redis.sort("people", SortArgs.Builder.by("w_*").get("o_*")))
             .containsExactly( "persistence", "cross-site", "security", null);
+
+      // SORT people BY w_* GET h_*->component
+      assertThat(redis.sort("people", SortArgs.Builder.by("w_*").get("h_*->component")))
+            .containsExactly( "persistence", "cross-site", "security", null);
+
       // SORT people BY w_* GET o_* GET #
       assertThat(redis.sort("people", SortArgs.Builder.by("w_*").get("o_*").get("#")))
             .containsExactly( "persistence", "ryan", "cross-site", "pedro",
                   "security", "tristan", null, "michael");
+      // SORT people BY w_* GET h_*->component GET #
+      assertThat(redis.sort("people", SortArgs.Builder.by("w_*").get("h_*->component").get("#")))
+            .containsExactly( "persistence", "ryan", "cross-site", "pedro",
+                  "security", "tristan", null, "michael");
       // SORT people BY w_* GET o GET #
       assertThat(redis.sort("people", SortArgs.Builder.by("w_*").get("o").get("#")))
             .containsExactly( null, "ryan", null, "pedro",
                   null, "tristan", null, "michael");
 
+      // FIXME: Operation against hash keys is a wrong type, should return null.
+      //  The operation succeeds if the storage type is Protostream since it can convert the bucket to byte[].
```

Comment:
This is a bit of a pain. If the SORT operation accesses an entry of the incorrect type, the operation shouldn't fail, it simply returns null for the value. This works fine with our default configuration with media-type to octet-stream.

The problem occurs when the user changes the cache's media type. Utilizing Protostream to values makes this operation succeed and silently returns incorrect values because we can serialize the bucket objects to bytes. I don't know of an easy way to fix this problem. I can think of a more robust approach adding a type hint into the cache entry metadata.